bfdd/bfd.c

```diff
@@ -216,6 +216,7 @@ void bfd_session_disable(struct bfd_session *bs)
 	bfd_echo_recvtimer_delete(bs);
 	bfd_xmttimer_delete(bs);
 	bfd_echo_xmttimer_delete(bs);
+	bs->vrf = NULL;
```

Comment:
Hi,
was there any null pointer crash happening when vrf was deleted ? It looks like when we will delete the VRF (which has an existing BFD session), the session will be disabled, but the old vrf pointer will still be hanging there in "struct bfd_session" variable.

Reply:
The problem was that the `vrf` pointer (in `struct bfd_session`) was `free`()d and wasn't reset. The right thing is to disable (so the session doesn't get handled anymore) and remove the invalid pointers.

This is exactly what the code was doing, but no one was removing the VRF pointer after it was deleted.

Here is the execution path:

*   `bfd_vrf_init` installs VRF callbacks - https://github.com/FRRouting/frr/blob/master/bfdd/bfd.c#L1729;
*   `bfd_vrf_disable` gets called on VRF teardown and `free`s the pointer - https://github.com/FRRouting/frr/blob/master/bfdd/bfd.c#L1697;
*   `bfd_session_disable` is called and disables all timers;